store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/handler/FederatedAddGraphHandlerParent.java

```diff
@@ -98,10 +100,11 @@ private void overwriteCacheProperty(final OP operation, final Store store) {
          * FederatedStore can't survive if a subgraph changes the static
          * cache to another cache, or re-initialises the cache.
          */
+        final String cacheServiceClass = isNull(store.getProperties()) ? null : store.getProperties().getCacheServiceClass(CACHE_SERVICE_CLASS_DEFAULT);
         final StoreProperties storeProperties = operation.getStoreProperties();
-        if (nonNull(storeProperties) && storeProperties.containsKey(CACHE_SERVICE_CLASS)) {
-            LOGGER.info(String.format("%s is removing %s from properties of the operation and substituting the FederatedStore's cache", this.getClass().getSimpleName(), CACHE_SERVICE_CLASS));
-            storeProperties.setCacheServiceClass(store.getProperties().getCacheServiceClass());
+        if (nonNull(storeProperties) && !storeProperties.getCacheServiceClass(cacheServiceClass).equals(cacheServiceClass)) {
+            LOGGER.info(String.format("Removing %s from properties of the operation and substituting the FederatedStore's cache", CACHE_SERVICE_CLASS));
```

Comment:
So this makes the subgraph use the same cache as the federated store? I thought this is what we were trying to avoid?

Reply:
Because it is still static all subgraphs must use the same static instance as the federated store. They now will use a suffix though to avoid collision